cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/common/utils.go

```diff
@@ -201,25 +206,50 @@ func NewClientGetter(configGetter core.KubernetesConfigGetter, cluster string) C
 // will be granted additional read privileges, we also need to ensure that the plugin can get a
 // config based on the service account rather than the request context
 func NewBackgroundClientGetter() ClientGetterFunc {
-	return func(ctx context.Context) (dynamic.Interface, apiext.Interface, error) {
+	return func(ctx context.Context) (kubernetes.Interface, dynamic.Interface, apiext.Interface, error) {
 		if config, err := rest.InClusterConfig(); err != nil {
-			return nil, nil, status.Errorf(codes.FailedPrecondition, "unable to get in cluster config due to: %v", err)
+			return nil, nil, nil, status.Errorf(codes.FailedPrecondition, "unable to get in cluster config due to: %v", err)
 		} else {
 			return clientGetterHelper(config)
 		}
 	}
 }
 
-func clientGetterHelper(config *rest.Config) (dynamic.Interface, apiext.Interface, error) {
+func clientGetterHelper(config *rest.Config) (kubernetes.Interface, dynamic.Interface, apiext.Interface, error) {
+	typedClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, nil, status.Errorf(codes.FailedPrecondition, fmt.Sprintf("unable to get typed client : %v", err))
+	}
 	dynamicClient, err := dynamic.NewForConfig(config)
 	if err != nil {
-		return nil, nil, status.Errorf(codes.FailedPrecondition, "unable to get dynamic client due to: %v", err)
+		return nil, nil, nil, status.Errorf(codes.FailedPrecondition, "unable to get dynamic client due to: %v", err)
 	}
 	apiExtensions, err := apiext.NewForConfig(config)
 	if err != nil {
-		return nil, nil, status.Errorf(codes.FailedPrecondition, "unable to get api extensions client due to: %v", err)
+		return nil, nil, nil, status.Errorf(codes.FailedPrecondition, "unable to get api extensions client due to: %v", err)
 	}
-	return dynamicClient, apiExtensions, nil
+	return typedClient, dynamicClient, apiExtensions, nil
+}
+
+// ref: https://blog.trailofbits.com/2020/06/09/how-to-check-if-a-mutex-is-locked-in-go/
+// I understand this is not really "kosher" in general for production usage,
+// but in one specific case (cache populateWith() func) it's okay as a sanity check
+// if it turns out not, I can always remove this check, it's not critical
+const mutexLocked = 1
+
+func RWMutexWriteLocked(rw *sync.RWMutex) bool {
+	// RWMutex has a "w" sync.Mutex field for write lock
+	state := reflect.ValueOf(rw).Elem().FieldByName("w").FieldByName("state")
+	return state.Int()&mutexLocked == mutexLocked
+}
```

Comment:
just a sanity check, same kind as you would, for example to check that the name of some object is not "", otherwise you raise an error. There is a piece of code that relies on some other piece of code having to lock the mutex for writing previously. So the former piece of code has this check. I mentioned this in the comment in NamespacedResourceWatcherCache.populateWith()